### PR TITLE
#10721 fix

### LINKED ISSUE #10721: [bug] beforeunload stops if there are unclosable tags into CKEditor
# 🐞 bug report
In EditView with a RTE field (CKEditor) containing some unclosable tags (img, input, br, hr, meta, etc.), window.onbeforeunload event stops exit.

### 🔥 How to trigger the error
beforeuloadckeditor

### 👎 Actual Behavior
window.onbeforeunload always stops exit

### 👍 Expected Behavior
If no changes was done, exit without alert

### 📷 Screenshot of configuration
[none]

### 📝 PHP/Apache/Nginx/Browser/CRM Logs
[independent]

### 🌍  Your Environment 
[independent]

### ❓ Additional context
[none]

### :heavy_check_mark: Fix
registerLeavePageWithoutSubmit() function in /public_html/layouts/basic/modules/Vtiger/resources/Edit.js:959 is called before CKEditor is ready: therefore InitialFormData gets different datas respect than when CKEditor completes initialization. eg: on page load, break tag is `<br>`; after CKEditor init, same tag will be `<br />`. Comparison always fails.
registerLeavePageWithoutSubmit() function have to be called after CKEditor init.
```code
...
registerLeavePageWithoutSubmit: function (form) {
  if(typeof CKEDITOR !== 'undefined' && typeof CKEDITOR.instances !== 'undefined' && Object.keys(CKEDITOR.instances).length) {
    CKEDITOR.on('instanceReady', function(e) {
      let InitialFormData = form.serialize();
      window.onbeforeunload = function (e) {
        if (InitialFormData != form.serialize() && form.data('submit') != "true") {
          return app.vtranslate("JS_CHANGES_WILL_BE_LOST");
        }
      };
    });
  } else {
    let InitialFormData = form.serialize();
    window.onbeforeunload = function (e) {
      if (InitialFormData != form.serialize() && form.data('submit') != "true") {
        return app.vtranslate("JS_CHANGES_WILL_BE_LOST");
      }
    };
  }
},
...
```

### PR DESCRIPTION
**Pull request by davide-alghi**
binds onbeforeunload handler after CKEditor init
inits all CKEditor instances: hidden ones too

fix #10721 issue

### CODE REVIEW

**Comment by davide-alghi** [comment]
I'm trying to figure out why my last two pull requests haven't completed the checks (**Some checks haven’t completed yet**): how can I solve the problems, if these are problems. Is the problem related to my fork? My branches? Pull request?
What I fear is all my next pull requests may have the same kind of problem.

File: public_html/layouts/basic/styles/Main.css [hunk 1]
```diff
@@ -28219,4 +28219,4 @@ td.fieldValue input, td.fieldValue .input-group {
     margin-bottom: 10px;
   }
 }
-/*# sourceMappingURL=Main.css.map */
\ No newline at end of file
+/*# sourceMappingURL=Main.css.map */
```
Comment (poradz) [inline public_html/layouts/basic/styles/Main.css]: ?

**Comment by poradz** [comment]
Hi!

> inits all CKEditor instances: hidden ones too

Please describe why :)

**Comment by davide-alghi** [comment]
> Hi!
> 
> > inits all CKEditor instances: hidden ones too
> 
> Please describe why :)

- we assume a RTE in a hidden block, with some unclosable tags: eg `<br>`
- on EditView page loading, serialize() get `<br>`
- now, you open the hidden block: CKEditor initializes textarea and changes `<br>` into `<br />`
- the comparison with the new serialize() fails

**Comment by davide-alghi** [comment]
> > I'm trying to figure out why my last two pull requests haven't completed the checks (**Some checks haven’t completed yet**): how can I solve the problems, if these are problems. Is the problem related to my fork? My branches? Pull request?
> > What I fear is all my next pull requests may have the same kind of problem.
> 
> Don't be afraid, you can keep sending Pull Requests.
> You introduce changes that require detailed analysis and tests and it takes time, we don't always have the time to do it right away and we have to wait until we have some free time.
> We are very grateful for your contributions, hope you keep improving the system.

Thanks Mariusz,
... of course I will continue to improve YF :+1:

**Pull request merged**